src/pages/App.tsx

```diff
@@ -80,97 +90,108 @@ export default function App(): ReactElement {
   }, [boot])
 
   return (
-    <Suspense fallback={null}>
-      <Web3ReactManager>
-        <MinichefProvider>
-          <GaugeProvider>
-            <TokensProvider>
-              <ExpandedPoolsProvider>
-                <UserStateProvider>
-                  <PricesAndVoteData>
-                    <PendingSwapsProvider>
-                      <AprsProvider>
-                        <RewardsBalancesProvider>
-                          <LocalizationProvider dateAdapter={AdapterDateFns}>
-                            <AppContainer>
-                              <TopMenu />
-                              <Switch>
-                                <Route exact path="/" component={Swap} />
-                                <Route exact path="/pools" component={Pools} />
-                                {pools.map((pool) => (
+    <QueryClientProvider client={queryClient}>
+      <ReactQueryDevtools initialIsOpen />
+      <Suspense fallback={null}>
+        <Web3ReactManager>
+          <MinichefProvider>
+            <GaugeProvider>
+              <TokensProvider>
+                <ExpandedPoolsProvider>
+                  <UserStateProvider>
+                    <PricesAndVoteData>
+                      <PendingSwapsProvider>
+                        <AprsProvider>
+                          <RewardsBalancesProvider>
+                            <LocalizationProvider dateAdapter={AdapterDateFns}>
+                              <AppContainer>
+                                <TopMenu />
+                                <Switch>
+                                  <Route exact path="/" component={Swap} />
                                   <Route
                                     exact
-                                    path={`/pools/${pool.poolName}/deposit`}
-                                    render={(props) => (
-                                      <Deposit
-                                        {...props}
-                                        poolName={pool.poolName}
-                                      />
-                                    )}
-                                    key={`${pool.poolName}-deposit`}
+                                    path="/pools"
+                                    component={Pools}
                                   />
-                                ))}
-                                {pools.map((pool) => (
+                                  {pools.map((pool) => (
```

Comment:
Rather than mapping over pools, can we just pass the poolName route param to the deposit or withdraw component? That way we'll have 2 routes instead of (n)

Reply:
big